### ISSUE TITLE
Add macFUSE module

### ISSUE THREAD

**Issue by 2gn** [LnL7/nix-darwin#668]
I'd like it to have an option to enable [macFUSE](https://osxfuse.github.io) ( FUSE filesystem for MacOS ).

**Comment by 2gn** [comment]
It seems it [doesn't](https://github.com/macfuse/macfuse/wiki/Getting-Started#notes).
> macFUSE does not require you to disable Gatekeeper or System Integrity Protection. All official macFUSE releases are digitally signed and approved by Apple's Notary Service. This means macFUSE is compatible with Gatekeeper and System Integrity Protection.

Latest macFUSE installs kernel extension to bridge user space and the filesystem.

ChatGPT told me that it used to require SIP to be disabled.

```
As of my knowledge cutoff in September 2021, disabling System Integrity Protection (SIP) was a common requirement for installing macFUSE on macOS. However, it's important to note that macOS updates and changes to the software may have altered this requirement. Therefore, I recommend consulting the official documentation and resources for macFUSE or checking with the developer or community support to confirm the current status and any specific installation instructions regarding SIP.
```

**Comment by andrewcrook** [comment]
Okay, but you still need to changes setting in recovery mode to allow third-party extensions , as you have noticed and the process may be different on older versions of macOS. 

Only way it might be possible to automated this is by using a profile under configurator and/or MDM tools these tend to be geared towards businesses and business IT management solutions. But yes you can make and install them (does require user input for security) 

However, profiles are not something you want to be downloading and installing off the net for obvious reasons e.g they can mess-up your Macs settings and can be a security risk.

**Comment by Enzime** [comment]
Leaving this open until it’s implemented

**Comment by andrewcrook** [comment]
@Enzime  I don't think it will ever be done because macOS simply doesn’t allow an automated way to reboot into recovery mode, login, bring up a security settings and allow system extensions and reboot again then install macFUSE. Perhaps I shouldn’t say 'doesn’t allow’ but rather there isn't a way

> macFUSE lets developers create file systems without having to write a single line of kernel code. This is achieved by running file system code in user space while the  **macFUSE kernel extension*** provides a "bridge" to the actual kernel interfaces.
[macFUSE ](https://macfuse.github.io)

> By default, ***third-party kernel extensions are disabled***  on Apple Silicon Macs. Click on the "Enable System Extensions…" button and enter your login password when prompted.
> Click on "Shut Down". After your Mac has shut down, press and hold the Touch ID or power button of your Mac to start the Recovery environment and launch Startup Security Utility.  
[macFUSE](https://github.com/macfuse/macfuse/wiki/Getting-Started)

**Comment by xelxebar** [comment]
Apparently usage of the FSKit backend gets around the main problem. Looks like it's fairly new with some considerable limitations, but it's worth keeping an eye on. The only fundamental limitation with FSKit is that mounts outside of /Volumes are not supported, but that's perhaps okay for us?

https://github.com/macfuse/macfuse/wiki/Getting-Started#fskit-backend

**Comment by andrewcrook** [comment]
Some time has passed since.  I found [FUSE-T](https://www.fuse-t.org) which doesn’t require a kernel extension by mounting a filing system as a NFS or SMB share. I am currently using [anylinuxfs](https://github.com/nohajc/anylinuxfs) which also mounts a filing system as a NFS share, however, everything is hidden in a micro Linux VM so you can theoretically use any file system compatible with Linux so FUSE Linux kernel support and its extensions, ext3/4, XFS, RAID, ZFS etc

**Comment by Enzime** [comment]
I started working on packaging FUSE-T at https://github.com/NixOS/nixpkgs/pull/348804
